packages/contracts/contracts/core/plugin/Plugin.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+import {_auth} from "../../utils/auth.sol";
+import {AppStorage} from "../../utils/AppStorage.sol";
+
+import {IDAO} from "../IDAO.sol";
+
+/// @title Plugin
+/// @notice An abstract, non-upgradeable contract to inherit from when creating a plugin being deployed via the `new` keyword.
+abstract contract Plugin is ERC165, Context, AppStorage {
+    bytes4 public constant PLUGIN_INTERFACE_ID = type(Plugin).interfaceId;
+
+    /// @notice Initializes the DAO by storing the associated DAO and registering the contract's [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID.
+    /// @dev This method is required to support [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822).
+    /// @param _dao The associated DAO address.
+    constructor(IDAO _dao) {
+        setDAO(address(_dao));
+    }
+
+    /// @notice A modifier to be used to check permissions on a target contract via the associated DAO.
+    /// @param _permissionId The permission identifier required to call the method this modifier is applied to.
+    modifier auth(bytes32 _permissionId) {
+        _auth(dao(), address(this), _msgSender(), _permissionId, _msgData());
```

Comment:
IMO i would name it `getDaoFromSlot()` or something of this sort, because it is essentially different from the `getDao()` we have in `DaoAuthorizable` and also tightly related to the subject raised related to the inheritance.

Reply:
If this tells us something meaningful from the perspective of the consumer, then let's go for it. But if it makes no difference, then I would name the same.

Reply:
amended